cnf-certification-test/lifecycle/scaling/crd_scaling.go

```diff
@@ -0,0 +1,219 @@
+			Version:  version.Name,
+			Resource: aCrd.Spec.Names.Plural,
+		}
+
+		logrus.Debugf("Looking for CRs from CRD: %s api version:%s group:%s plural:%s", aCrd.Name, version.Name, aCrd.Spec.Group, aCrd.Spec.Names.Plural)
+		crs, err := clients.DynamicClient.Resource(gvr).List(context.TODO(), metav1.ListOptions{})
+
+		if err != nil {
+			logrus.Errorf("error getting %s: %v\n", aCrd.Name, err)
+			return false
+		}
+		for _, item := range crs.Items {
+			cr := item
+			resource := schema.GroupResource{
+				Group:    aCrd.Spec.Group,
+				Resource: aCrd.Spec.Names.Plural,
+			}
+			namespace := cr.Object["metadata"].(map[string]interface{})["namespace"].(string)
+			name := cr.Object["metadata"].(map[string]interface{})["name"].(string)
+
+			scalefiled, err := clients.ScalesGetter.Scales(namespace).Get(context.TODO(), resource, name, metav1.GetOptions{})
+			if err != nil {
+				return false
+			}
+
+			tnf.ClaimFilePrintf(namespace)
+			if hpa, ok := CheckHpa(hpaList, name, namespace, aCrd.Spec.Names.Kind); ok {
+				TestScaleHPACrd(scalefiled, hpa, timeout)
+				continue
+			}
+
+			replicas = scalefiled.Spec.Replicas
+
+			if replicas <= 1 {
+				// scale up
+				replicas++
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, true) {
+					logrus.Error("can't scale cr =")
+					return false
+				}
+				// scale down
+				replicas--
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, false) {
+					logrus.Error("can't scale cr =")
+					return false
+				}
+			} else {
+				// scale down
+				replicas--
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, false) {
+					logrus.Error("can't scale cr =")
+					return false
+				} // scale up
+				replicas++
+				if !scaleCrsHelper(clients.ScalesGetter, resource, scalefiled, replicas, true) {
+					logrus.Error("can't scale cr =")
+					return false
+				}
+			}
+			return true
+		}
+	}
+
+	return true
+}
+
+func scaleCrsHelper(scalesGetter scale.ScalesGetter, rc schema.GroupResource, autoscalerpram *scalingv1.Scale, replicas int32, up bool) bool {
+	if up {
+		logrus.Trace("scale UP CRS to ", replicas, " replicas ")
+	} else {
+		logrus.Trace("scale DOWN CRS to ", replicas, " replicas ")
+	}
+
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+		namespace := autoscalerpram.GetNamespace()
+		scalefiled, err := scalesGetter.Scales(namespace).Get(context.TODO(), rc, autoscalerpram.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		scalefiled.Spec.Replicas = replicas
+		_, err = scalesGetter.Scales(namespace).Update(context.TODO(), rc, scalefiled, metav1.UpdateOptions{})
+		if err != nil {
+			logrus.Error("can't update DynamicClient ")
+			return err
+		}
+		return nil
+	})
+	if retryErr != nil {
+		logrus.Error("can't scale DynamicClient ", " error=", retryErr)
+		return false
+	}
+	return true
```

Comment:
Shouldn't we call another get here to check that the status has updated to match the spec update? Same for the HPA helper.